### ISSUE TITLE
[button] Disabled button can be focused & activated when using a screen reader

### ISSUE THREAD

**Issue by Hzunax** [ing-bank/lion#722]
## Describe the bug

Disabled buttons can be focused & activated by navigating through the form controls with a screen reader.

## To Reproduce

- Go to the lion demo page on https://lion-web-components.netlify.app/?path=/docs/buttons-button--disabled
- Open your screen reader (e.g. VoiceOver on Mac)
- Navigate to the disabled button (without using the tab key)
- The screen reader will read out "<label>, button" instead of the expected "<label>, dimmed, button"
- Activate the button
- The screen reader will read out "press <label>, button" instead of the expected "dimmed"

## Expected behavior

Same behavior as with a disabled `<button>` element. Users should not be able to activate it.

## Screenshots

Expected behavior on focus:
![image](https://user-images.githubusercontent.com/23220825/82463176-c651cc00-9abc-11ea-82bc-7d4b40284c8a.png)

Actual behavior on focus:
![image](https://user-images.githubusercontent.com/23220825/82462781-517e9200-9abc-11ea-9a2d-b9d14524d349.png)

Expected behavior on activate:
![image](https://user-images.githubusercontent.com/23220825/82462926-7a9f2280-9abc-11ea-8a01-704c587bb571.png)

Actual behavior on activate:
![image](https://user-images.githubusercontent.com/23220825/82463046-a0c4c280-9abc-11ea-8910-98412eb0db5f.png)

## Proposed solution

Make the `lion-button` an actual `button` element instead of a `div`.

Or add the attribute `aria-disabled="true"` when `disabled` is true.
I have tried the latter in the browser's dev tools and it gave the expected behavior. ([more info on aria-disabled](https://a11y-101.com/development/aria-disabled))

## Additional context

Tested with macOS built-in VoiceOver application on

- Chrome
- Firefox
- Safari

**Comment by jorenbroekema** [comment]
When I go to these demos on Chrome with MacOS built-in VoiceOver, I cannot navigate to disabled buttons whatsoever, so nothing will be read out by the VO for disabled buttons.

Can you explain how you are navigating to the button?

**Comment by Hzunax** [comment]
When using the "tab" button to navigate, it is indeed skipped. However, if I use the VO keys (Ctrl + Option in my case) with the arrow buttons, I can navigate to the button.

You can also use VoiceOver's rotor (VO keys + U) and navigate with the arrows to the "Form Controls" and then to "Disabled button"

Another way to do so is by navigating through all form controls. This can be done with VO keys + Cmd + J.

**Comment by Hzunax** [comment]
On a side note, the "hello" button you see in the screenshots is not present in the storybook demo. I simply added it for demonstration purposes.

Open the browser dev tools and add the following to the page's HTML:
`<button disabled>Hello</button>`

**Comment by tlouisse** [comment]
Thanks for testing this @Hzunax We can probably fix this here by syncing the aria-disabled attr with the disabled state: https://github.com/ing-bank/lion/blob/master/packages/core/src/DisabledMixin.js#L48

**Comment by Hzunax** [comment]
That would indeed work. However, not all components that extend from the DisabledMixin seem to need the `aria-disabled` attribute. Elements that natively support the `disabled` attribute, don't need the `aria-disabled`, similar to how they don't need to set `tabindex="-1"`. 
So to avoid redundant use of aria, I was thinking instead of syncing with `DisabledMixin`, I could either sync it with `DisabledWithTabIndexMixin` or abstracting it into its own Mixin. 
Or just set the attribute in the `updated` lifecycle of the LionSwitchButton and LionButton (which seem to be the only affected components).
